### PR TITLE
Fix Fuse SDK with local cache

### PR DESCRIPTION
**Pull request by LuQQiu**
Fixes https://github.com/Alluxio/alluxio/issues/16793
Set "FileId" in UfsBaseFileSystem to fulfill the local cache requirement.

Enable FUSE SDK tests with different cache combinations.

### CODE REVIEW

**Comment by jja725** [comment]
Is it possible to add some tests to cover this?

**Comment by LuQQiu** [comment]
@jja725 Looks like there are more issues using FUSE SDK with local cache, would want to understand and fix all the related problems and then add the tests to give good test coverage.

Currently we have comprehensive tests covering FUSE SDK with local/S3, but do not with different cache combinations (local data cache / local metadata cache), will add that part as well after testing, understanding, and debugging phrases.

**Comment by LuQQiu** [comment]
But good point, may be i should start by at lease enabling original tests all with data cache/metadata cache combinations

File: core/client/fs/src/main/java/alluxio/client/file/ufs/UfsBaseFileSystem.java [hunk 1]
```diff
@@ -399,6 +399,7 @@ private URIStatus transformStatus(UfsStatus ufsStatus) {
         CommonUtils.stripPrefixIfPresent(ufsStatus.getName(), mRootUFS.getPath())));
     FileInfo info = new FileInfo().setName(ufsUri.getName())
         .setPath(ufsStatus.getName())
+        .setFileId(ufsUri.hashCode())
```
Comment (huanghua78) [inline core/client/fs/src/main/java/alluxio/client/file/ufs/UfsBaseFileSystem.java:402]: Some of the fields may be optional, but the FileId should be a must.
We should check and assert this in build().

Comment (LuQQiu) [reply core/client/fs/src/main/java/alluxio/client/file/ufs/UfsBaseFileSystem.java:402]: Yeah FileInfo is better to be a builder with arguments check to force set needed fields.
But currently FileInfo allows field modification for serialization and deserialization.

Need to implement a customized serializer that supports the new fileInfo without setters.
Prestodb has examples of how to implement it.
I previously consider change it but it's a too big of change that i didn't get time to. And this is one item on the 3.0 best practice list.

**Comment by LuQQiu** [comment]
```
19:31:06.500 [INFO] Running alluxio.client.file.cache.LocalCacheFileInStreamTest
Error: 7.145 [ERROR] Tests run: 52, Failures: 0, Errors: 12, Skipped: 0, Time elapsed: 0.633 s <<< FAILURE! - in alluxio.client.file.cache.LocalCacheFileInStreamTest
Error: 7.149 [ERROR] alluxio.client.file.cache.LocalCacheFileInStreamTest.readEmptyFileThroughReadByteBuffer[0: page_size(1,048,576), in_stream_buffer_size(0)]  Time elapsed: 0.005 s  <<< ERROR!
java.lang.IllegalArgumentException: to be read content must be equal or shorter than target buffer length
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at alluxio.client.file.cache.LocalCacheFileInStream.readInternal(LocalCacheFileInStream.java:229)
	at alluxio.client.file.cache.LocalCacheFileInStream.read(LocalCacheFileInStream.java:133)
	at alluxio.client.file.cache.LocalCacheFileInStreamTest.readEmptyFileThroughReadByteBuffer(LocalCacheFileInStreamTest.java:160)
```

**Comment by LuQQiu** [comment]
@huanghua78 @jja725 @beinan 
Fix the File id with local cache issue.
Added a bunch of unit/integration tests with cache combinations to make sure no easy to occur correctness issues

**Comment by LuQQiu** [comment]
Passed with S3 in Linux/Mac
mvn -DfailIfNoTests=false -Dtest=alluxio.fuse.ufs.\*Test test -Dmaven.javadoc.skip -Dfindbugs.skip -Dcheckstyle.skip -Dalluxio.test.s3a.path=s3://<>/ -Dlicense.skip -Ds3a.accessKeyId=<> -Ds3a.secretKey=<>  -pl \!webui

**Review by jja725 — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**